### DISCUSSION TITLE
I would like to customize the data before it's saved during a soft delete operation.

### DISCUSSION THREAD

**Discussion by bzy107** [filamentphp/filament#9070] [Help] [answered]
I was able to customize the data during creation and editing, as shown below.
Is it possible to do the same for deletion? 

Here is the link for reference:
https://filamentphp.com/docs/3.x/panels/resources/creating-records#customizing-data-before-saving

**Comment by sinnbeck** [comment]
How do you need to customize it? Generally you would probably use an event listener to catch the delete event

https://laravel.com/docs/10.x/eloquent#events

**Reply by sinnbeck** [reply]
If you are talking about an action button you can hook into the before event perhaps.

```php
DeleteAction::make()
                    ->before(fn($record) => dd($record)) //do stuff before deletion
```
But this wont work for bulk deleting without hooking into that as well.
https://filamentphp.com/docs/3.x/tables/actions#bulk-actions

**Reply by bzy107** [reply]
I want to update the updated_user_id column with the target user when it's deleted.

**Reply by sinnbeck** [reply, marked as answer]
Then you should be able to use either. But as this sounds to be session dependent I would probably add it to the action like in my second answer.

```php
DeleteAction::make()
    ->before(fn($record) => $record->update(['updated_user_id' => auth()->id()])

**Reply by bzy107** [reply]
Thank you.

I have resolved it!!

**Reply by sinnbeck** [reply]
Happy to help :)
